substrate/runtime/staking/src/lib.rs

```diff
@@ -381,6 +494,114 @@ impl<T: Trait> Module<T> {
 				.collect::<Vec<_>>()
 		);
 	}
+
+	/// Lookup an T::AccountIndex to get an Id, if there's one there.
+	pub fn lookup_index(index: T::AccountIndex) -> Option<T::AccountId> {
+		let usize_index: usize = As::<usize>::as_(index);
+		let mut set = Self::enum_set(T::AccountIndex::sa(usize_index / ENUM_SET_SIZE));
+		let i = usize_index % ENUM_SET_SIZE;
+		if i < set.len() {
+			Some(set.swap_remove(i))
+		} else {
+			None
+		}
+	}
+
+	/// `true` if the account `index` is ready for reclaim.
+	pub fn can_reclaim(index: T::AccountIndex) -> bool {
+		let try_index: usize = index.as_();
+		let try_set = Self::enum_set(T::AccountIndex::sa(try_index / ENUM_SET_SIZE));
+		let item_index = try_index % ENUM_SET_SIZE;
+		item_index < try_set.len() && Self::voting_balance(&try_set[item_index]).is_zero()
```

Comment:
`.is_zero()` or `< existential deposit`?

Reply:
VotingBalance should not be compared directly to ExistentialDeposit. VotingBalance == FreeBalance + ReservedBalance, neither of which may ever be allowed to be set s.t. >0 && <ExistentialDeposit. By extension, `VotingBalance` is never < ExistentialDeposit, so `.is_zero()` works fine.